go-controller/pkg/metrics/master_test.go

```diff
@@ -0,0 +1,360 @@
+//go:build !race
+
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ovn-org/libovsdb/client"
+	"github.com/ovn-org/libovsdb/model"
+	"github.com/ovn-org/libovsdb/ovsdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdbops"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	libovsdbtest "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/testing/libovsdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	"github.com/prometheus/client_golang/prometheus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	fakeclientgo "k8s.io/client-go/kubernetes/fake"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+)
+
+type fakeHistogramInterface interface {
+	prometheus.Metric
+	prometheus.Collector
+	Observe(float64)
+}
+
+type fakeHistogram struct {
+	prometheus.Metric
+	prometheus.Collector
+	observeCalledFunc func(v float64)
+}
+
+func (f fakeHistogram) Observe(v float64) {
+	f.observeCalledFunc(v)
+}
+
+func getHistoMock() (fakeHistogramInterface, *int, *float64) {
+	f := fakeHistogram{}
+	var numberOfTimesCalled int
+	var totalAmountAdded float64
+	f.observeCalledFunc = func(v float64) {
+		numberOfTimesCalled += 1
+		totalAmountAdded += v
+	}
+	return f, &numberOfTimesCalled, &totalAmountAdded
+}
+
+func setupOvn(nbData libovsdbtest.TestSetup) (client.Client, client.Client, *libovsdbtest.Cleanup) {
+	nbClient, sbClient, cleanup, err := libovsdbtest.NewNBSBTestHarness(nbData)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	return sbClient, nbClient, cleanup
+}
+
+func getKubeClient(nodeCount int) kube.Kube {
+	var nodes []corev1.Node
+	for i := 0; i < nodeCount; i++ {
+		nodes = append(nodes, corev1.Node{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("nodeName-%d", i),
+			},
+		})
+	}
+	kubeFakeClient := fakeclientgo.NewSimpleClientset(&corev1.NodeList{Items: nodes})
+	fakeClient := &util.OVNClientset{
+		KubeClient: kubeFakeClient,
+	}
+	return kube.Kube{fakeClient.KubeClient, nil, nil, nil}
+}
+
+func setHvCfg(nbClient client.Client, hvCfg int, hvCfgTimestamp time.Time) {
+	nbGlobal, err := libovsdbops.FindNBGlobal(nbClient)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	ops, err := nbClient.Where(nbGlobal).Mutate(nbGlobal, model.Mutation{
+		Field:   &nbGlobal.HvCfg,
+		Mutator: ovsdb.MutateOperationAdd,
+		Value:   hvCfg,
+	},
+		model.Mutation{
+			Field:   &nbGlobal.HvCfgTimestamp,
+			Mutator: ovsdb.MutateOperationAdd,
+			Value:   int(hvCfgTimestamp.UnixMilli()),
+		})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	gomega.Expect(ops).To(gomega.HaveLen(1))
+	_, err = libovsdbops.TransactAndCheck(nbClient, ops)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+}
+
+var _ = ginkgo.Describe("Config duration recorder operations", func() {
+	ginkgo.Context("recordConfigDuration()", func() {
+		ginkgo.It("should record a result for kind metric", func() {
```

Comment:
will it make sense to merge this and the next test and verify correct recoring with a single test?